agent/service_manager.go

```diff
@@ -379,39 +379,76 @@ func mergeServiceConfig(defaults *structs.ServiceConfigResponse, service *struct
 	if ns.Proxy.MeshGateway.Mode == structs.MeshGatewayModeDefault {
 		ns.Proxy.MeshGateway.Mode = defaults.MeshGateway.Mode
 	}
+	if !ns.Proxy.TransparentProxy {
+		ns.Proxy.TransparentProxy = defaults.TransparentProxy
+	}
+
+	// seenUpstreams stores the upstreams seen from the local registration so that we can also add synthetic entries.
```

Comment:
```suggestion
	// remoteUpstreams stores the upstreams seen from the local registration so that we can also add synthetic entries.
```

Reply:
Hmm, the variable used to be called `seenUpstreams`, but below it is `remoteUpstreams`. The comment here though leans in the direction of this probably being a local thing not a remote thing so I'm unclear which it's supposed to be.

Reply:
Should this comment actually move down to the `localUpstreams` variable below?

Reply:
Yep, I must have moved that map but missed moving the comment.